### PR TITLE
Rename licence -> license throughout codebase

### PR DESCRIPTION
**Pull request by jom**
Built off of #2552 

While working on #2552, I realized we have a very brittle setup with the usage of `licence` vs (US) `license`. This finally standardizes it to use `license` and migrates the licenses appropriately. Our strings should previously all been updated, but this switches over the code/options.

### Changes proposed in this Pull Request

* Migrate `licence` to `license` in code and options.

### Testing instructions

* On `trunk`, activate licenses for products.
* Switch to this branch (note: needs an asset rebuild) and ensure licenses aren't lost. Try activating, deactivating, etc and see how it goes.

### Deprecated Code

* `WP_Job_Manager_Helper::has_licenced_products` -> `WP_Job_Manager_Helper::has_licensed_products`
* `WP_Job_Manager_Helper::get_plugin_licence` -> `WP_Job_Manager_Helper::get_plugin_license`
* `WP_Job_Manager_Helper::licence_output` -> `WP_Job_Manager_Helper::licence_output`
* `WP_Job_Manager_Helper::activate_licence` -> `WP_Job_Manager_Helper::activate_license`
* `WP_Job_Manager_Helper::deactivate_licence` -> `WP_Job_Manager_Helper::deactivate_license`

### CODE REVIEW

File: includes/helper/class-wp-job-manager-helper.php
Comment (jom) [inline includes/helper/class-wp-job-manager-helper.php:904]: This is changed from the API's `request` method.

File: includes/helper/class-wp-job-manager-helper.php
Comment (aaronfc) [inline includes/helper/class-wp-job-manager-helper.php]: Wrong placeholders: missing `$$` at the end. `$$next-version$$`

File: includes/helper/class-wp-job-manager-helper-api.php
Comment (aaronfc) [inline includes/helper/class-wp-job-manager-helper-api.php]: Very minor thing, but I think it makes more sense to handle this "translation" first thing in the method.

Currently `$defaults` are defined above and they reference the old name (`licence_key`).

File: includes/helper/class-wp-job-manager-helper-options.php
Comment (aaronfc) [inline includes/helper/class-wp-job-manager-helper-options.php:140]: We are exposing the '_version' key through `get_license_option()` method. This could, potentially, lead to some "delete", "get" or "update" (which use `get_license_option`) for the '_version' key as product slug (if it's not sanitized before).

Not sure it's worth the effort trying to fix this. WDYT?

**Review by aaronfc — COMMENTED** [review]
Code looks good! (This is a brave change 😨 🔥 )

I have left some comments and have reviewed the code in detail. I will do tests later.

File: includes/helper/class-wp-job-manager-helper-api.php
Comment (jom) [reply includes/helper/class-wp-job-manager-helper-api.php]: Moved in 72413d56682e60d9ce555e4cb1c15e97ad1fe879

File: includes/helper/class-wp-job-manager-helper.php
Comment (jom) [reply includes/helper/class-wp-job-manager-helper.php]: Good catch! Fixed in 7042a172fdfeee109c71be58d50ac2c5ad80ba17.

File: includes/helper/class-wp-job-manager-helper-options.php
Comment (jom) [reply includes/helper/class-wp-job-manager-helper-options.php:140]: I added some simple checking in 21dc369aff8c697ea41a87753c0773297006209e. WDYT?

Comment (aaronfc) [reply includes/helper/class-wp-job-manager-helper-options.php:140]: Good enough!

**Review by aaronfc — APPROVED** [review]
Working like a charm 🔥

**Comment by jom** [comment]
@aaronfc Thanks for the review! I'll hold off merging until the parent PR is merged.

**Pull request merged**
